### DISCUSSION TITLE
URL used to clone a repo through  git

### DISCUSSION THREAD

**Discussion by TheSupremeGod** [community/community#22859] [New to GitHub]
<p>So I have always used this style to clone my own repos from bash:</p>
<p>git clone <a href="https://github.com/User/Repo" rel="nofollow noopener">https://github.com/User/Repo</a></p>
<p>Sometimes, though I see someone post a different URL used to clone a repo:</p>
<p>git clone <a href="https://github.com/User/Repo.git" rel="nofollow noopener">https://github.com/User/Repo.git</a></p>
<p>Is there any difference in the URL? Does GitHub assume the .git ending if it is not specified?</p>

**Comment by ThirdEyeRose** [comment, marked as answer]
<p>Hi <a class="mention" href="/thesupremegod">@thesupremegod</a>,</p>
<p>You are correct. While the standard Git URL ends in .git, cloning with the repository URL works because GitHub understands what the Git operation is trying to do when it reaches the controller.</p>
<p>Hope that helps. :slight_smile:</p>
